### ISSUE TITLE
MacOS Build Error: Qt 5 and 6 Simultaneously Installed.

### ISSUE THREAD

**Issue by da2ce7** [bitcoin/bitcoin#25947]
This error seems to appear when `brew install qt@5` is installed at the same time as: `brew install qt`. (ie, qt version 6):

``` bash
% brew list --version | grep qt

qt 6.3.1_1
qt@5 5.15.5_1
```

```
Making all in src
  CXX      qt/libbitcoinqt_a-bantablemodel.o
In file included from qt/bantablemodel.cpp:14:
In file included from /usr/local/Cellar/qt@5/5.15.5_1/lib/QtCore.framework/Headers/QLocale:1:
/usr/local/Cellar/qt@5/5.15.5_1/lib/QtCore.framework/Headers/qlocale.h:1028:25: error: unknown type name 'QStringRef'; did you mean 'CFStringRef'?
    short toShort(const QStringRef &s, bool *ok = nullptr) const;
                        ^~~~~~~~~~
                        CFStringRef
/usr/local/include/QtCore/qdebug.h:531:1: note: 'CFStringRef' declared here
Q_FORWARD_DECLARE_CF_TYPE(CFString);
^
/usr/local/include/QtCore/qglobal.h:802:77: note: expanded from macro 'Q_FORWARD_DECLARE_CF_TYPE'
#  define Q_FORWARD_DECLARE_CF_TYPE(type) typedef const struct __ ## type * type ## Ref
                                                                            ^
<scratch space>:112:1: note: expanded from here
CFStringRef
^
```

```bash
% git clean -ffxd
% git rev-parse HEAD
e191fac4f3c37820f0618f72f0a8e8b524531ab8
% ./autogen.sh
% ./configure
% make
```
output: https://gist.github.com/da2ce7/c697259e204c961876ad7f7057e95b34

config.log: https://gist.github.com/da2ce7/ec7276de7d0284681cd73b0cdcb90f3b

brew list --version --formulae: https://gist.github.com/da2ce7/25d6473224a1d69a28cf24e44abb55b7

_Originally posted by @da2ce7 in https://github.com/bitcoin/bitcoin/issues/21901#issuecomment-1229561087_

**Comment by hebasto** [comment]
@da2ce7 

> This error seems to appear when `brew install qt@5` is installed at the same time as: `brew install qt`.

This is unsupported configuration. See: https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md#qt 

Could be fixed after #25797.

**Comment by fanquake** [comment]
There's more to this than just qt5 and qt6 being installed via brew. I've tested building on a macOS (M1) system, with both installed, and there are no compilation problems.

**Comment by hebasto** [comment]
I was going to fix this issue, and already had a draft of the patch, but...

macOS Monterey 12.6 (21G115, Intel):
```
% brew list --version | grep qt
qt 6.3.1_3
qt@5 5.15.5_2
```

Having the same result as @fanquake:
> there are no compilation problems.

Definitely, I observed problems in the past, but not now.

So, @da2ce7 and other devs, who observe such an issue, please provide more details about your systems to let us to reproduce it. Also try to update `brew`'s packages.

**Comment by hebasto** [comment]
A fix suggested in https://github.com/bitcoin/bitcoin/pull/26056#issuecomment-1242510118.
